File: ChallengeBackend/src/main/java/br/com/savedra/challengebackend/repository/CampaignRepository.java
package br.com.savedra.challengebackend.repository;

import br.com.savedra.challengebackend.model.Campaign;
import org.springframework.data.mongodb.repository.MongoRepository;
import java.util.List;

public interface CampaignRepository extends MongoRepository<Campaign, String> {
    List<Campaign> findBySegment(String segment);
}

File: app/src/main/java/br/com/savedra/challengecrm/data/repository/CampaignRepository.kt
package br.com.savedra.challengecrm.data.repository

import br.com.savedra.challengecrm.model.Campaign
import com.google.firebase.firestore.FirebaseFirestore
import kotlinx.coroutines.tasks.await

class CampaignRepository {
  private val firestore = FirebaseFirestore.getInstance()

  suspend fun sendCampaign(campaign: Campaign) {
    firestore.collection("campaigns").add(campaign).await()
  }

  suspend fun getCampaigns(userSegment: String? = null): List<Campaign> {
    val query = if (userSegment != null) {
        firestore.collection("campaigns").whereIn("segment", listOf(userSegment, "Todos"))
    } else {
        firestore.collection("campaigns")
    }
    val result = query.get().await()
    return result.map { document ->
        document.toObject(Campaign::class.java)
    }
  }
}